checks.d/varnish.py

```diff
@@ -88,16 +72,47 @@ def check(self, instance):
             tags = []
         else:
             tags = list(set(tags))
+        varnishstat_path = instance.get("varnishstat")
         name = instance.get('name')
 
+        # Get version and version-specific args from varnishstat -V.
+        version, use_xml = self._get_version_info(varnishstat_path)
+
+        # Parse metrics from varnishstat.
+        arg = '-x' if use_xml else '-1'
+        cmd = [varnishstat_path, arg]
+
+        if name is not None:
+            cmd.extend(['-n', name])
+            tags += [u'varnish_name:%s' % name]
+        else:
+            tags += [u'varnish_name:default']
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                     stderr=subprocess.PIPE)
+        output, error = proc.communicate()
+        if error and len(error) > 0:
+            self.log.error(error)
+        self._parse_varnishstat_metrics(varnishstat_path, use_xml, tags)
```

Comment:
you should pass "output" instead of "varnishstat_path" here. 
See pullrequest https://github.com/DataDog/dd-agent/pull/1377